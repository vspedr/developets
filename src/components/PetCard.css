.card-image {
  object-fit: cover;
  width: 100%;
  height: 320px;
}

.ui.card,
.ui.cards > .card {
  width: 320px;
}
