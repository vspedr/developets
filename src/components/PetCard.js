import React, { useState } from 'react';

import placeholder from '../assets/img/paw.png';

import {
  Container,
  Content,
  Description,
  GitHubIcon,
  GitHubSection,
  Owner,
  SubTitle,
  Thumbnail,
  Title,
} from '../styles/CardStyle';

export const PetCard = ({ pet }) => {
  const [imageFallbackError, setImageFallbackError] = useState(false);
  const { name, type, description, img, owner } = pet;

  const imgSrc = imageFallbackError ? placeholder : img;

  const handleImageError = () => {
    setImageFallbackError(true);
  };

  const thumbnail = (
    img
      ? <Thumbnail alt={ name } src={ imgSrc } onError={ handleImageError } />
      : <Thumbnail alt={ name } />
  );

  return (
    <Container>
      { thumbnail }
      <Content>
        <Title>{ name }</Title>
        <SubTitle>{ `the ${type}` }</SubTitle>
        <Description>{ description }</Description>
      </Content>
      <GitHubSection href={ owner ? `http://github.com/${owner}` : null }>
        <GitHubIcon />
        <Owner>{ owner || null }</Owner>
      </GitHubSection>
    </Container>
  );
};