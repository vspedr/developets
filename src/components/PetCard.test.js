import React from 'react';
import { shallow } from "enzyme";

import { PetCard } from './PetCard';

describe("PetCard", () => {
  it("Snapshot", () => {
    const rendered = shallow(<PetCard />);
    expect(rendered.html()).toMatchSnapshot();
  });
});
