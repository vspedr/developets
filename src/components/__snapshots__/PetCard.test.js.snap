// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetCard Snapshot 1`] = `"<div class=\\"ui card\\"><img src=\\"paw.png\\" class=\\"ui image card-image\\"/><div class=\\"content\\"><div class=\\"meta\\">the undefined</div></div></div>"`;
